compiler/packages/babel-plugin-react-compiler/src/Inference/InferMutationAliasingEffects.ts

```diff
@@ -644,6 +653,13 @@ function applyEffect(
           if (DEBUG) {
             console.log('apply function expression effects');
           }
+          applyEffect(
+            context,
+            state,
+            {kind: 'MutateTransitiveConditionally', value: effect.function},
+            aliased,
+            effects,
+          );
```

Comment:
even if we're able to construct a signature for a local function expression, we still need to consider it as mutating to account for mutations of its captures.